### PR TITLE
Update mathcomp packages

### PR DESCRIPTION
**Pull request by CohenCyril**
ci-skip: coq-mathcomp-ssreflect.2.3.0 coq-mathcomp-fingroup.2.3.0 coq-mathcomp-algebra.2.3.0 coq-mathcomp-solvable.2.3.0 coq-mathcomp-field.2.3.0

### CODE REVIEW

File: released/packages/coq-mathcomp-algebra/coq-mathcomp-algebra.2.3.0/opam [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+description: """
+This library contains definitions and theorems about discrete
+(i.e. with decidable equality) algebraic structures : ring, fields,
+ordered fields, real fields,  modules, algebras, integers, rational
+numbers, polynomials, matrices, vector spaces...
+"""
+
+url {
+src: "https://github.com/math-comp/math-comp/archive/mathcomp-2.3.0.tar.gz"
+checksum: "sha256=dba29e4019e3086ee17e6f5babdfd03c5c437697a613a32dc155b4665fd344ff"
```
Comment (palmskog) [inline released/packages/coq-mathcomp-algebra/coq-mathcomp-algebra.2.3.0/opam:63]:
```suggestion
checksum: "sha256=19e13c8765007f95b4656d8902bc66e10b072ab94ab51031c5efb860827d05ec"
```
Also elsewhere.

**Comment by palmskog** [comment]
@CohenCyril [this](https://gitlab.inria.fr/coq/opam/-/jobs/5012513) looks like an actual version incompatibility:
```
  - install coq                    8.18.0   [required by coq-mathcomp-ssreflect]
  - install atdts                  2.15.0   [required by elpi]
  - install atdgen                 2.15.0   [required by elpi]
  - install elpi                   1.17.4   [required by coq-mathcomp-ssreflect]
  - install coq-elpi               1.19.3   [required by coq-hierarchy-builder]
  - install coq-hierarchy-builder  1.6.0    [required by coq-mathcomp-ssreflect]
  - install coq-mathcomp-ssreflect 2.3.0
 
# COQC order.v
# COQC binomial.v
# File "./order.v", line 2082, characters 0-2:
# Error: Syntax error: illegal begin of vernac.
# 
# make[2]: *** [Makefile.coq:838: order.vo] Error 1
# make[2]: *** [order.vo] Deleting file 'order.glob'
```

File: released/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.2.3.0/opam [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+dev-repo: "git+https://github.com/math-comp/math-comp.git"
+license: "CECILL-B"
+
+build: [ make "-C" "mathcomp/ssreflect" "-j" "%{jobs}%" ]
+install: [ make "-C" "mathcomp/ssreflect" "install" ]
+depends: [
+  "coq" {(>= "8.18" & < "8.21~") | (= "dev")}
+  # Please keep the "dev" above as it is required for the coq-dev Docker images
+  "elpi" {>= "1.17.0"}
+  "coq-hierarchy-builder" { >= "1.5.0"}
```
Comment (palmskog) [inline released/packages/coq-mathcomp-ssreflect/coq-mathcomp-ssreflect.2.3.0/opam:15]:
```suggestion
  "coq-hierarchy-builder" { >= "1.7.0"}
```

**Comment by palmskog** [comment]
@erikmd MathComp-Docker images with 2.3.0 should now be straightforward.

**Comment by erikmd** [comment]
Yes, thanks a lot @palmskog for the ping 👍

I'm looking after this

**Pull request merged**